src/api/externalUser.ts

```diff
@@ -303,13 +319,27 @@ export class ExternalUser {
 
             const result = await UserApps.verifyRequests(requests);
 
-            const loginRequest = result.find((r) => r.getPayload().origin === window.location.origin)?.getPayload();
+            const loginRequest = result.find((r) => {
+                if (r.getType() === 'LoginRequest') {
+                    const payload = r.getPayload() as LoginRequestPayload;
+
+                    if (payload && payload.origin === window.location.origin) {
+                        return true;
+                    }
+                }
+
+                return false;
+            });
             const keyFromStorage = await keyManager.getKey({ level: KeyManagerLevel.BROWSER_LOCAL_STORAGE });
 
             if (!loginRequest) throwError('No login request found for this origin', SdkErrors.OriginMismatch);
 
-            if (loginRequest.publicKey.toString() !== keyFromStorage.toString()) {
-                throwError('Key in request does not match', SdkErrors.KeyNotFound);
+            if (loginRequest.getType() === 'LoginRequest') {
+                const payload = loginRequest.getPayload() as LoginRequestPayload;
```

Comment:
instead of using `as`, you could do
`const payload = new LoginRequest(loginRequest).getPayload()`

Reply:
this way we KNOW we are future proofed for type safety